Add new entry about prototype model in SDLC (#1999)
Codecademy/docs: content/general/concepts/software-development-life-cycle/terms/prototype/prototype.md

---
Title: 'Prototype Model'
Description: 'The prototype model involves creating a simplified model of the product to test its concept and functionality.'
Subjects:
  - 'Mobile Development'
  - 'Web Development'
Tags:
  - 'Development'
  - 'Models'
CatalogContent:
  - 'learn-how-to-code'
  - 'paths/full-stack-engineer-career-path'
---

The **prototype model** is a software development approach that involves creating a simplified model of the future product to test its concept and functionality with minimal resource costs. The process of creating a prototype involves a repeating cycle of steps, including development, testing, and continuous improvement based on testing results and feedback. The prototype may include only the essential design elements and features that are necessary for visualizing and demonstrating the fundamental capabilities of the product. Prototypes are used by the development team in collaboration with customers and stakeholders to test the product's design, architecture, functionality, and other aspects.

## Objectives of Prototype Model

> 

Reviewer: @onekir

5.  Convincing stakeholders: Prototypes are an effective way to demonstrate a product's potential to investors, clients, or management, which can provide necessary support and financing.

## Types of Prototypes

Prototypes can be classified by the method of creation:

1. Paper prototype: the simplest and fastest to create, requiring only pen and paper. It’s great for initial versions of prototypes.  Paper prototypes show the structure and design of the product, as well as its basic functional elements.
2. Digital prototype: created in specialized software and services for prototyping. Digital prototypes more accurately show the product functionality and may contain additional elements, such as animations and interactive components.

Prototypes can also be classified by the level of detail:

1.  Low-fidelity prototype: reflects the product basic structure and functionality of the product, but does not include detailed design and interface elements.
2.  High-fidelity prototype: a detailed and interactive prototype that includes interface design and other elements that will be present in the final product.

## Stages in the Prototype Model

> 1.  Requirements gathering. At this stage, the project team communicates with customers, conducts research, and collects all the requirements for the future product.
> 2.  Design. At this stage, the structure and design of the future product are determined. For example, for a website, this would include the number of pages, the purpose of each page, and the placement of key elements.
> 3.  Creating a prototype using the structure developed in the previous stage.
> 4.  Prototype testing and feedback collection. The developed prototype is tested to identify errors and areas for improvement.
> 5.  Prototype refinement. The feedback obtained from testing is used to improve the prototype and more fully meet the requirements of the product.
> 6.  Development of the final product. After the prototype has been refined and meets customer requirements, the team moves on to the final product development.

Reviewer: ```suggestion
1.  Requirements gathering: At this stage, the project team communicates with customers, conducts research, and collects all the requirements for the future product.
2.  Design: At this stage, the structure and design of the product are determined. For example, for a website, this would include the number of pages, the purpose of each page, and the placement of key elements.
3.  Prototyping: Creating a prototype using the design that was developed in the previous stage.
4.  Prototype testing and feedback collection: The developed prototype is tested to identify errors and areas for improvement.
5.  Prototype refinement: The feedback obtained from testing is used to improve the prototype and meet the requirements of the product.
6.  Development: After the prototype has been refined and meets customer requirements, the team moves on to the final product development.
```
